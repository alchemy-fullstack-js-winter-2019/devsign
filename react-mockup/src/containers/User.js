import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import UserInfo from '../components/user/UserInfo';
import { getUserHandle, getUserProfilePicture } from '../selectors/session';

class User extends PureComponent {
  static propTypes = {
    handle: PropTypes.string.isRequired,
    profilePicture: PropTypes.string.isRequired
  }

  render() {
    return (
      <UserInfo {...this.props} />
    );
  }
}

const mapStateToProps = state => ({
  handle: getUserHandle(state),
  profilePicture: getUserProfilePicture(state)
});

export default connect(
  mapStateToProps
)(User);
