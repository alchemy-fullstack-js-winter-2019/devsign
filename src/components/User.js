import React from 'react';
import styles from './User.css';

function User() {
  return (
    <main className={styles.User}>

    </main>
  );
}
export default User;
