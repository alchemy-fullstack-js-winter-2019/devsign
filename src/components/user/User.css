.User {
  article {
    text-align: center;
  } 
  figure {
    margin: .5em;
  }
  img {
    border-radius: 60px;
    margin: 0;
  }
  h3 {
    margin: .5em;
  }
  h2{
    margin: .5em;
  }
}