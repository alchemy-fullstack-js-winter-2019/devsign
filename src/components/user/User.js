import React from 'react';
import PropTypes from 'prop-types';
import styles from '../css/User.css';

function User({ user }) {
  const { handle, profileImg } = user;
  return ( 
    <figure className={styles.user}>
      <figcaption>{handle}</figcaption>
      <img alt="profileImage" src={profileImg} />
    </figure>
  );
}

User.propTypes = {
  user: PropTypes.object.isRequired
};

export default User;
