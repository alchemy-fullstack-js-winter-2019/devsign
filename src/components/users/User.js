import React from 'react';
import PropTypes from 'prop-types';
import Tweets from '../tweets/Tweets';

function User({ img, username, bio, tweets }) {
  return (
    <main>
      <img src={img}/>
      <p>{username}</p>
      <p>{bio}</p>
      <Tweets tweets={tweets} />
    </main>
  );
}

User.propTypes = {
  img: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  bio: PropTypes.string.isRequired,
  tweets: PropTypes.array.isRequired
};

export default User;
