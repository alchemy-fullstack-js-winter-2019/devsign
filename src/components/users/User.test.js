import React from 'react';
import renderer from 'react-test-renderer';
import User from '../users/User';
import { MemoryRouter } from 'react-router-dom';

describe('User', () => {
  it('displays a user', () => {
    const user = {
      id: 1,
      img: 'image.com',
      username: 'user2',
      bio: 'I am a person; I swear',
      tweets: [
        {
          id: 1,
          userId: 1,
          img: 'https://cdn.shopify.com/s/files/1/1061/1924/products/Thinking_Face_Emoji_large.png?v=1480481060',
          text: 'twitter you twit'
        },
        {
          id: 2,
          userId: 1,
          img: 'https://cdn.shopify.com/s/files/1/1061/1924/products/Thinking_Face_Emoji_large.png?v=1480481060',
          text: 'twitter you twit twit'
        },
        {
          id: 3,
          userId: 1,
          img: 'https://cdn.shopify.com/s/files/1/1061/1924/products/Thinking_Face_Emoji_large.png?v=1480481060',
          text: 'twitter you twit twit twitter twit'
        }
      ]
    };

    const tree = renderer.create(
      <MemoryRouter>
        <User {...user} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
