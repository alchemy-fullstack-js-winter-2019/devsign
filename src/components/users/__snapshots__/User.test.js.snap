// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User displays a user 1`] = `
<main>
  <img
    src="image.com"
  />
  <p>
    user2
  </p>
  <p>
    I am a person; I swear
  </p>
  <div>
    <h2>
      Latest Tweets:
    </h2>
    <ul
      className="Tweets"
    >
      <li>
        <a
          href="/user/1"
          onClick={[Function]}
        >
          <img
            src="https://cdn.shopify.com/s/files/1/1061/1924/products/Thinking_Face_Emoji_large.png?v=1480481060"
          />
          <p>
            twitter you twit
          </p>
        </a>
      </li>
      <li>
        <a
          href="/user/1"
          onClick={[Function]}
        >
          <img
            src="https://cdn.shopify.com/s/files/1/1061/1924/products/Thinking_Face_Emoji_large.png?v=1480481060"
          />
          <p>
            twitter you twit twit
          </p>
        </a>
      </li>
      <li>
        <a
          href="/user/1"
          onClick={[Function]}
        >
          <img
            src="https://cdn.shopify.com/s/files/1/1061/1924/products/Thinking_Face_Emoji_large.png?v=1480481060"
          />
          <p>
            twitter you twit twit twitter twit
          </p>
        </a>
      </li>
    </ul>
  </div>
</main>
`;
